src/Data/List/Base.agda

```diff
@@ -144,65 +144,60 @@ sum = foldr _+_ 0
 product : List ℕ → ℕ
 product = foldr _*_ 1
 
-length : ∀ {a} {A : Set a} → List A → ℕ
+length : List A → ℕ
 length = foldr (const suc) 0
 
 ------------------------------------------------------------------------
 -- Operations for constructing lists
 
-[_] : ∀ {a} {A : Set a} → A → List A
+[_] : A → List A
 [ x ] = x ∷ []
 
-fromMaybe : ∀ {a} {A : Set a} → Maybe A → List A
+fromMaybe : Maybe A → List A
 fromMaybe (just x) = [ x ]
 fromMaybe nothing  = []
 
-replicate : ∀ {a} {A : Set a} → ℕ → A → List A
+replicate : ℕ → A → List A
 replicate zero    x = []
 replicate (suc n) x = x ∷ replicate n x
 
-inits : ∀ {a} {A : Set a} → List A → List (List A)
+inits : List A → List (List A)
 inits []       = [] ∷ []
 inits (x ∷ xs) = [] ∷ map (x ∷_) (inits xs)
 
-tails : ∀ {a} {A : Set a} → List A → List (List A)
+tails : List A → List (List A)
 tails []       = [] ∷ []
 tails (x ∷ xs) = (x ∷ xs) ∷ tails xs
 
 -- Scans
 
-scanr : ∀ {a b} {A : Set a} {B : Set b} →
-        (A → B → B) → B → List A → List B
+scanr : (A → B → B) → B → List A → List B
 scanr f e []       = e ∷ []
 scanr f e (x ∷ xs) with scanr f e xs
 ... | []     = []                -- dead branch
 ... | y ∷ ys = f x y ∷ y ∷ ys
 
-scanl : ∀ {a b} {A : Set a} {B : Set b} →
-        (A → B → A) → A → List B → List A
+scanl : (A → B → A) → A → List B → List A
 scanl f e []       = e ∷ []
 scanl f e (x ∷ xs) = e ∷ scanl f (f e x) xs
 
-module _ {a} {A : Set a} where
-
 -- Tabulation
 
-  applyUpTo : (ℕ → A) → ℕ → List A
-  applyUpTo f zero    = []
-  applyUpTo f (suc n) = f zero ∷ applyUpTo (f ∘ suc) n
+applyUpTo : (ℕ → A) → ℕ → List A
+applyUpTo f zero    = []
+applyUpTo f (suc n) = f zero ∷ applyUpTo (f ∘ suc) n
 
-  applyDownFrom : (ℕ → A) → ℕ → List A
-  applyDownFrom f zero = []
-  applyDownFrom f (suc n) = f n ∷ applyDownFrom f n
+applyDownFrom : (ℕ → A) → ℕ → List A
+applyDownFrom f zero = []
+applyDownFrom f (suc n) = f n ∷ applyDownFrom f n
 
-  tabulate : ∀ {n} (f : Fin n → A) → List A
-  tabulate {zero}  f = []
-  tabulate {suc n} f = f Fin.zero ∷ tabulate (f ∘ Fin.suc)
+tabulate : ∀ {n} (f : Fin n → A) → List A
```

Comment:
Couldn't you drop `∀ {n}` and it means the same thing? I believe that then you can also drop `n = ` from the lhs below.

Reply:
Yup we could. However the conclusion of #697 seems to be that until we have a mechanism in emacs for displaying a term's full type at the site of its definition, we're going to stick with only generalising things of type `Level` and `Set` in order to avoid the user having hold to much in their head at once. This policy will probably be relaxed later to include things like naturals.